test/test_profiler.py

```diff
@@ -138,6 +140,182 @@ def test_json_output():
+            assert frame_field in speedscope_frame_list[frame_index], (
+                f"Field named '{frame_field}' not in Speedscope output"
+                f"at 'shared.frames[{frame_index}]'"
+            )
+        frame_index_name = speedscope_frame_list[frame_index]["name"]
+        assert frame_index_name == function_name, (
+            f"Speedscope output 'shared.frames[{frame_index}].name' has value "
+            f"{frame_index_name} != {function_name}"
+        )
+
+    speedscope_profile_list = output["profiles"]
+    assert len(speedscope_profile_list) == 1
+    speedscope_profile = speedscope_profile_list[0]
+    speedscope_profile_fields = ("type", "name", "unit", "startValue", "endValue", "events")
+    for profile_field in speedscope_profile_fields:
+        assert (
+            profile_field in speedscope_profile
+        ), f"Field named '{profile_field}' not in Speedscope output at 'profiles[0]'"
+
+    # speedscope_profile["endValue"] is not tested because a fake_time mock
+    # timer is used to replace time.time, and this mock causes session.duration
+    # to differ from self._event_time just before exiting SpeedscopeRenderer.render
+    start_time_in_seconds = 0.0
+    event_time_in_seconds = start_time_in_seconds
+    assert speedscope_profile["type"] == "evented"
+    assert speedscope_profile["unit"] == "seconds"
+    assert speedscope_profile["startValue"] == start_time_in_seconds
+
+    """Two helper functions for constructing a timeline and testing it"""
+
+    def update_event_time_on_sleep_close(time_increment_in_seconds: float):
+        """Updates running total of event time in seconds and returns updated value"""
+        nonlocal event_time_in_seconds
+        event_time_in_seconds += time_increment_in_seconds
+        return event_time_in_seconds
+
+    def get_approx_abs_tol(time_in_seconds: float):
+        """Computes absolute tolerance for given time value to account for CI time jitter"""
+        uncertainty_per_quarter_second = 0.1
+        absolute_tolerance = (time_in_seconds / 0.25) * uncertainty_per_quarter_second
+        return absolute_tolerance
+
+    speedscope_event_list = speedscope_profile["events"]
+    sleep_a_time_in_seconds = 0.25
+    sleep_b_time_in_seconds = 0.5
+
+    output_event_tuple = (
+        SpeedscopeEvent(
+            SpeedscopeEventType.OPEN,
+            event_time_in_seconds,
+            distinct_functions_called["test_speedscope_output"],
+        ),
+        SpeedscopeEvent(
+            SpeedscopeEventType.OPEN,
+            event_time_in_seconds,
+            distinct_functions_called["long_function_a"],
+        ),
+        SpeedscopeEvent(
+            SpeedscopeEventType.OPEN,
+            event_time_in_seconds,
+            distinct_functions_called["sleep"],
+        ),
+        SpeedscopeEvent(
+            SpeedscopeEventType.CLOSE,
+            update_event_time_on_sleep_close(sleep_a_time_in_seconds),
```

Comment:
Personally, I'd prefer to keep tests pretty explicit. I think it's okay to hardcode things in tests, rather than computing stuff like this.

Reply:
So here I'd just put literal `0`, or `sleep_a_time_in_seconds`, or `sleep_a_time_in_seconds + sleep_b_time_in_seconds`

Reply:
I replaced those calls with literals and removed the local variables re: sleep time.

Reply:
Replaced function calls with literals in [`97925fb2`](https://github.com/joerick/pyinstrument/pull/160/commits/97925fb284a34a111b7b99bbef8cd0bb72e7c584).